src/apps/eos/actions/__init__.py

```diff
@@ -0,0 +1,107 @@
+from apps.eos import consts, updaters, helpers
+from apps.eos.actions import layout
+
+from trezor.messages.EosTxActionRequest import EosTxActionRequest
+from trezor.utils import HashWriter
+from trezor.crypto.hashlib import sha256
+
+async def process_action(ctx, sha, action):
+    if not check_action(action):
+        raise ValueError("Unknown action")
+
+    updaters.hashupdate_action_common(sha, action.common)
+    if action.buy_ram is not None:
```

Comment:
a better approach seems to be to make this switch based on `action.common.name` and pick out the respective data field (as opposed to checking which data field is present and generating that action)

Reply:
done

Reply:
@matejcik Hey. I have reworked that part according to your suggestion. And also fixed the styling issues according to checks during build. Everything is squashed and re-based. Thanks!